leetcodePython/BinarySearch/bs_1648.py

```diff
@@ -0,0 +1,25 @@
+import heapq
+from typing import List
+
+
+class Solution:
+    def arithmeticSum(self, a_n: int, a_1: int, n: int) -> int:
+        return (a_n + a_1 + 1) * n // 2
+
+    def maxProfit(self, inventory: List[int], orders: int) -> int:
+        inventory.sort(reverse=True) # inventory high to low
+        inventory += [0]
+        ans = 0
+        greater_element_count = 1
+        for i in range(len(inventory)-1):
+            if inventory[i] > inventory[i+1]:
+                n = inventory[i] - inventory[i+1]
+                if greater_element_count*n < orders:
+                    ans += greater_element_count * self.arithmeticSum(inventory[i], inventory[i+1], n)
+                    orders -= greater_element_count*n
```

Comment:
Consider `[4, 4, 2]`

For each `4` to become `2` it needs arithmetic sum, we have `two` of `4`, we count the `two` by using `greater_element_count`